tests/theseus_tests/core/test_gnc_cost.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import pytest
+
+import theseus as th
+
+import torch
+from tests.core.common import BATCH_SIZES_TO_TEST
+
+
+def _new_gnc_robust_cf(
```

Comment:
Is this test suite the same as the one for the base class? If so, I think it's better to reuse the original test code and just add some control flow that uses `GNCRobustCostFunction` and any extra args when the loss is of type `GNCRobustLoss`.